### PR TITLE
mgr/cephadm: track daemon deployment post actions by daemon name instead of type

### PR DESCRIPTION
**Pull request by Daniel-Pivonka**
in rare cases tracking by type can cause a daemon to be missed when doing post actions if it is not in the cache yet

Fixes: https://tracker.ceph.com/issues/52866
Signed-off-by: Daniel Pivonka <dpivonka@redhat.com>

### CODE REVIEW

File: src/pybind/mgr/cephadm/serve.py [hunk 1]
```diff
@@ -1004,8 +1004,15 @@ def _check_daemons(self) -> None:
 
         # do daemon post actions
         for daemon_type, daemon_descs in daemons_post.items():
-            if daemon_type in self.mgr.requires_post_actions:
-                self.mgr.requires_post_actions.remove(daemon_type)
+            run_post = False
+
+            x = ''
+
```
Comment (mgfritch) [inline src/pybind/mgr/cephadm/serve.py:1010]:
```suggestion
```

typo?

File: src/pybind/mgr/cephadm/serve.py [hunk 2]
```diff
@@ -1004,8 +1004,15 @@ def _check_daemons(self) -> None:
...
+            for d in daemon_descs:
+                if d.name() in self.mgr.requires_post_actions:
+                    self.mgr.requires_post_actions.remove(d.name())
```
Comment (mgfritch) [inline src/pybind/mgr/cephadm/serve.py:1013]: what if the post action has a transient failure? remove the daemon name iff the post action succeeds?

File: src/pybind/mgr/cephadm/serve.py (hunk 1 above)
Comment (Daniel-Pivonka) [reply src/pybind/mgr/cephadm/serve.py]: oops ya that a left over from debugging

**Comment by adk3798** [comment]
http://pulpito.front.sepia.ceph.com/adking-2021-10-30_00:36:30-rados:cephadm-wip-adk-testing-2021-10-29-1557-distro-basic-smithi/

1 Failure Tracked by https://tracker.ceph.com/issues/49287

File: src/pybind/mgr/cephadm/serve.py (hunk 2 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/serve.py:1010]: how do we deal with non-transient errors then?

Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/serve.py:1010]: maybe a thing for another PR?

**Review by sebastian-philipp — APPROVED** [review]

**Pull request merged**